diff -ru GrafX2-2.3-original/src/misc.c GrafX2-2.3/src/misc.c
--- GrafX2-2.3-original/src/misc.c	2011-02-06 18:16:48.000000000 +0100
+++ GrafX2-2.3/src/misc.c	2011-07-11 00:37:11.000000000 +0200
@@ -787,7 +787,7 @@
   len = sizeof(maxmem);
   sysctl(mib,2,&maxmem,&len,NULL,0);
   return maxmem;
-#elif defined(__BEOS__) || defined(__HAIKU__) || defined(__SKYOS__) || defined(__amigaos4__) || defined(__TRU64__)
+#elif defined(__BEOS__) || defined(__HAIKU__) || defined(__SKYOS__) || defined(__amigaos4__) || defined(__TRU64__) || defined(__SYLLABLE__)
   // No <sys/sysctl.h> on BeOS or Haiku
   // AvailMem is misleading on os4 (os4 caches stuff in memory that you can still allocate)
 #warning "There is missing code there for your platform ! please check and correct :)"
