diff -ur cups-1.2.1_orig/backend/ieee1284.c cups-1.2.1/backend/ieee1284.c
--- cups-1.2.1_orig/backend/ieee1284.c	2006-04-19 21:36:43.000000000 +0000
+++ cups-1.2.1/backend/ieee1284.c	2007-02-28 20:34:16.000000000 +0000
@@ -51,12 +51,16 @@
  */
 
 #ifndef SNMP_BACKEND
-#  ifdef __linux
+#  if defined(__linux) || defined(__SYLLABLE__)
 #    include <sys/ioctl.h>
-#    include <linux/lp.h>
+#    ifdef __linux
+#      include <linux/lp.h>
+#    else
+#      include <atheos/lp.h>
+#    endif
 #    define IOCNR_GET_DEVICE_ID		1
 #    define LPIOC_GET_DEVICE_ID(len)	_IOC(_IOC_READ, 'P', IOCNR_GET_DEVICE_ID, len)
-#  endif /* __linux */
+#  endif /* __linux || __SYLLABLE__ */
 
 #  ifdef __sun
 #    ifdef __sparc
@@ -85,9 +89,9 @@
 	manufacturer[256],		/* Manufacturer string */
 	serial_number[1024];		/* Serial number string */
   int	manulen;			/* Length of manufacturer string */
-#ifdef __linux
+#if defined(__linux) || defined(__SYLLABLE__)
   int	length;				/* Length of device ID info */
-#endif /* __linux */
+#endif /* __linux || __SYLLABLE__ */
 #if defined(__sun) && defined(ECPPIOC_GETDEVID)
   struct ecpp_device_id did;		/* Device ID buffer */
 #endif /* __sun && ECPPIOC_GETDEVID */
@@ -121,7 +125,7 @@
   * Get the device ID string...
   */
 
-#ifdef __linux
+#if defined(__linux) || defined(__SYLLABLE__)
   if (!ioctl(fd, LPIOC_GET_DEVICE_ID(device_id_size), device_id))
   {
    /*
@@ -157,7 +161,7 @@
   else
     printf("get_device_id: ioctl failed - %s\n", strerror(errno));
 #  endif /* DEBUG */
-#endif /* __linux */
+#endif /* __linux || __SYLLABLE__ */
 
 #if defined(__sun) && defined(ECPPIOC_GETDEVID)
   did.mode = ECPP_CENTRONICS;
diff -ur cups-1.2.1_orig/backend/parallel.c cups-1.2.1/backend/parallel.c
--- cups-1.2.1_orig/backend/parallel.c	2006-04-10 02:51:12.000000000 +0000
+++ cups-1.2.1/backend/parallel.c	2007-02-28 20:34:16.000000000 +0000
@@ -110,9 +110,9 @@
 #if defined(HAVE_SIGACTION) && !defined(HAVE_SIGSET)
   struct sigaction action;		/* Actions for POSIX signals */
 #endif /* HAVE_SIGACTION && !HAVE_SIGSET */
-#ifdef __linux
+#if defined(__linux) || defined(__SYLLABLE__)
   unsigned int	status;			/* Port status (off-line, out-of-paper, etc.) */
-#endif /* __linux */
+#endif /* __linux || __SYLLABLE__ */
 
 
  /*
@@ -269,7 +269,7 @@
 
   paperout = 0;
 
-#if defined(__linux) && defined(LP_POUTPA)
+#if ( defined(__linux) || defined(__SYLLABLE__) ) && defined(LP_POUTPA)
  /*
   * Show the printer status before we send the file...
   */
@@ -295,7 +295,7 @@
 
     sleep(5);
   }
-#endif /* __linux && LP_POUTPA */
+#endif /* ( __linux || __SYLLABLE__ ) && LP_POUTPA */
 
  /*
   * Now that we are "connected" to the port, ignore SIGTERM so that we
@@ -453,7 +453,7 @@
 				/* Funky hex numbering used for some devices */
 #endif /* __hpux || __sgi || __sun */
 
-#ifdef __linux
+#if defined(__linux) || defined(__SYLLABLE__)
   int	i;			/* Looping var */
   int	fd;			/* File descriptor */
   char	device[255],		/* Device filename */
@@ -466,6 +466,8 @@
     strcpy(basedevice, "/dev/parallel/");
   else if (!access("/dev/printers/", 0))
     strcpy(basedevice, "/dev/printers/");
+  else if(!access("/dev/printer/lp/", 0))
+    strcpy(basedevice, "/dev/printer/lp/");
   else
     strcpy(basedevice, "/dev/lp");
 
diff -ur cups-1.2.1_orig/backend/usb-unix.c cups-1.2.1/backend/usb-unix.c
--- cups-1.2.1_orig/backend/usb-unix.c	2006-05-20 12:23:40.000000000 +0000
+++ cups-1.2.1/backend/usb-unix.c	2007-02-28 20:36:25.000000000 +0000
@@ -45,7 +45,6 @@
 
 int	open_device(const char *uri);
 
-
 /*
  * 'print_device()' - Print a file to a USB device.
  */
@@ -76,9 +75,9 @@
 #if defined(HAVE_SIGACTION) && !defined(HAVE_SIGSET)
   struct sigaction action;		/* Actions for POSIX signals */
 #endif /* HAVE_SIGACTION && !HAVE_SIGSET */
-#ifdef __linux
+#if defined(__linux) || defined(__SYLLABLE__)
   unsigned int	status;			/* Port status (off-line, out-of-paper, etc.) */
-#endif /* __linux */
+#endif /* __linux || __SYLLABLE__ */
 
   (void)argc;
   (void)argv;
@@ -162,7 +161,7 @@
 
   paperout = 0;
 
-#if defined(__linux) && defined(LP_POUTPA)
+#if ( defined(__linux) || defined(__SYLLABLE__) ) && defined(LP_POUTPA)
  /*
   * Show the printer status before we send the file...
   */
@@ -188,7 +187,7 @@
 
     sleep(5);
   }
-#endif /* __linux && LP_POUTPA */
+#endif /* ( __linux || __SYLLABLE__ ) && LP_POUTPA */
 
  /*
   * Now that we are "connected" to the port, ignore SIGTERM so that we
@@ -341,7 +340,7 @@
 void
 list_devices(void)
 {
-#ifdef __linux
+#if defined(__linux) || defined(__SYLLABLE__)
   int	i;			/* Looping var */
   int	fd;			/* File descriptor */
   char	format[255],		/* Format for device filename */
@@ -359,8 +358,10 @@
     strcpy(format, "/dev/usblp%d");
   else if (!access("/dev/usb/usblp0", 0))
     strcpy(format, "/dev/usb/usblp%d");
-  else
+  else if (!access("/dev/usb/lp0", 0))
     strcpy(format, "/dev/usb/lp%d");
+  else
+    strcpy(format, "/dev/printer/usb/%d");
 
  /*
   * Then open each USB device...
@@ -445,7 +446,7 @@
   */
 
   if (!strncmp(uri, "usb:/dev/", 9))
-#ifdef __linux
+#if defined(__linux)
     return (-1); /* Do not allow direct devices anymore */
   else if (!strncmp(uri, "usb://", 6))
   {
@@ -471,8 +472,10 @@
       strcpy(format, "/dev/usblp%d");
     else if (!access("/dev/usb/usblp0", 0))
       strcpy(format, "/dev/usb/usblp%d");
-    else
+    else if (access("/dev/usb/lp0", 0) == 0)
       strcpy(format, "/dev/usb/lp%d");
+    else
+      strcpy(format, "/dev/printer/usb/%d");
 
    /*
     * Then find the correct USB device...
diff -ur cups-1.2.1_orig/backend/usb.c cups-1.2.1/backend/usb.c
--- cups-1.2.1_orig/backend/usb.c	2006-04-06 20:03:32.000000000 +0000
+++ cups-1.2.1/backend/usb.c	2007-02-28 20:34:35.000000000 +0000
@@ -72,7 +72,7 @@
 
 #ifdef __APPLE__
 #  include "usb-darwin.c"
-#elif defined(__linux) || defined(__sun) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__)
+#elif defined(__linux) || defined(__sun) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__) || defined(__SYLLABLE__)
 #  include "usb-unix.c"
 #else
 /*
