diff -ru gcc-3.4.3-original/Makefile.in gcc-3.4.3/Makefile.in
--- gcc-3.4.3-original/Makefile.in	2004-11-05 04:14:06.000000000 +0000
+++ gcc-3.4.3/Makefile.in	2005-04-20 23:28:59.000000000 +0000
@@ -23418,7 +23418,8 @@
 	esac; \
 	$(SET_LIB_PATH) \
 	echo "$$msg"; \
-	cd gcc && $(MAKE) $(GCC_FLAGS_TO_PASS) $$compare
+	cd gcc
+	#&& $(MAKE) $(GCC_FLAGS_TO_PASS) $$compare
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}` ; export s; \
 	$(SET_LIB_PATH) \
diff -ru gcc-3.4.3-original/configure.in gcc-3.4.3/configure.in
--- gcc-3.4.3-original/configure.in	2004-11-04 17:05:21.000000000 +0000
+++ gcc-3.4.3/configure.in	2005-04-20 23:28:59.000000000 +0000
@@ -696,6 +697,9 @@
     ;;
   sparc-*-solaris* | sparc64-*-solaris* | sparcv9-*-solaris*)
     ;;
+  *-*-syllable)
+    noconfigdirs="$noconfigdirs target-newlib"
+    ;;
   v810-*-*)
     noconfigdirs="$noconfigdirs bfd binutils gas gcc gdb ld target-libstdc++-v3 opcodes target-libgloss ${libgcj}"
     ;;
@@ -1170,6 +1174,9 @@
   *-*-linux*)
     target_makefile_frag="config/mt-linux"
     ;;
+  *-*-syllable*)
+	target_makefile_frag="config/mt-syllable"
+	;;
   *-*-aix4.[[3456789]]* | *-*-aix[[56789]].*)
     # nm and ar from AIX 4.3 and above require -X32_64 flag to all ar and nm 
     # commands to handle both 32-bit and 64-bit objects.  These flags are
diff -ru gcc-3.4.3-original/gcc/config.gcc gcc-3.4.3/gcc/config.gcc
--- gcc-3.4.3-original/gcc/config.gcc	2004-09-08 15:16:11.000000000 +0000
+++ gcc-3.4.3/gcc/config.gcc	2005-04-20 23:28:59.000000000 +0000
@@ -938,6 +938,16 @@
 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h svr4.h i386/beos-elf.h"
 	extra_parts='crtbegin.o crtend.o'
 	;;
+i[34567]86-*-syllable*)
+	xmake_file=x-syllable
+	xm_defines="HAVE_ATEXIT POSIX BSTRING"
+	tmake_file='t-slibgcc-syllable i386/t-syllable i386/t-crtstuff'
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h svr4.h i386/syllable.h"
+	extra_parts='crtbegin.o crtend.o crtbeginS.o crtendS.o'
+	gnu_ld=yes
+	float_format=i386
+	thread_file='syllable'
+	;;
 i[34567]86-*-freebsd2 | i[34567]86-*-freebsd2.* | i[34567]86-*-freebsd*aout*)
 	tm_file="${tm_file} i386/unix.h i386/bsd.h i386/gas.h i386/gstabs.h i386/freebsd-aout.h"
 	tmake_file=t-freebsd
diff -ru gcc-3.4.3-original/gcc/configure gcc-3.4.3/gcc/configure
--- gcc-3.4.3-original/gcc/configure	2004-11-05 04:14:06.000000000 +0000
+++ gcc-3.4.3/gcc/configure	2005-04-20 23:28:59.000000000 +0000
@@ -11760,13 +11760,13 @@
   fi
   rm -f conftest.* conftest[123].*
 fi
-if test x$gcc_cv_ld_ro_rw_mix = xread-write; then
-
-cat >>confdefs.h <<\_ACEOF
+#if test x$gcc_cv_ld_ro_rw_mix = xread-write; then
+#
+#cat >>confdefs.h <<\_ACEOF
 #define HAVE_LD_RO_RW_SECTION_MIXING 1
-_ACEOF
+#_ACEOF
 
-fi
+#fi
 echo "$as_me:$LINENO: result: $gcc_cv_ld_ro_rw_mix" >&5
 echo "${ECHO_T}$gcc_cv_ld_ro_rw_mix" >&6
 
diff -ru gcc-3.4.3-original/gcc/gthr.h gcc-3.4.3/gcc/gthr.h
--- gcc-3.4.3-original/gcc/gthr.h	2002-12-16 18:19:38.000000000 +0000
+++ gcc-3.4.3/gcc/gthr.h	2005-04-20 23:28:59.000000000 +0000
@@ -85,6 +85,8 @@
 #include "gthr-dce.h"
 #elif _SOLARIS_THREADS
 #include "gthr-solaris.h"
+#elif defined __SYLLABLE__
+#include "gthr-syllable.h"
 
 /* Include GTHREAD_FILE if one is defined.  */
 #elif defined(HAVE_GTHR_DEFAULT)
diff -ru gcc-3.4.3-original/libstdc++-v3/configure gcc-3.4.3/libstdc++-v3/configure
--- gcc-3.4.3-original/libstdc++-v3/configure	2004-10-28 21:52:03.000000000 +0000
+++ gcc-3.4.3/libstdc++-v3/configure	2005-04-20 23:28:59.000000000 +0000
@@ -3926,6 +3926,11 @@
     ;;
   esac
   ;;
+
+syllable*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 esac
 
 fi
diff -ru gcc-3.4.3-original/libtool.m4 gcc-3.4.3/libtool.m4
--- gcc-3.4.3-original/libtool.m4	2004-05-18 09:08:38.000000000 +0000
+++ gcc-3.4.3/libtool.m4	2005-04-20 23:28:59.000000000 +0000
@@ -734,6 +734,10 @@
     ;;
   esac
   ;;
+syllable*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 esac
 ])
 file_magic_cmd=$lt_cv_file_magic_cmd
diff -ru gcc-3.4.3-original/ltconfig gcc-3.4.3/ltconfig
--- gcc-3.4.3-original/ltconfig	2004-03-05 21:05:42.000000000 +0000
+++ gcc-3.4.3/ltconfig	2005-04-20 23:28:59.000000000 +0000
@@ -1348,6 +1348,22 @@
   need_version=yes
   ;;
 
+syllable*)
+  version_type=linux
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='${libname}${release}.so$versuffix ${libname}${release}.so$major $libname.so'
+  soname_spec='${libname}${release}.so$major'
+  shlibpath_var=LIBRARY_PATH
+  shlibpath_overrides_runpath=no
+  # This implies no fast_install, which is unacceptable.
+  # Some rework will be needed to allow for fast_install
+  # before this can be enabled.
+  hardcode_into_libs=yes
+  dynamic_linker='Syllable ld.so'
+  can_build_shared=yes
+  ;;
+
 sysv4 | sysv4.2uw2* | sysv4.3* | sysv5*)
   version_type=linux
   library_names_spec='${libname}${release}.so$versuffix ${libname}${release}.so$major $libname.so'
diff -ru gcc-3.4.3-original/ltmain.sh gcc-3.4.3/ltmain.sh
--- gcc-3.4.3-original/ltmain.sh	2004-02-08 01:37:28.000000000 +0000
+++ gcc-3.4.3/ltmain.sh	2005-04-20 23:28:59.000000000 +0000
@@ -2650,6 +2650,11 @@
 	  verstring="$verstring:${current}.0"
 	  ;;
 
+	syllable)
+	  major=.`expr $current - $age`
+	  versuffix="$major.$age.$revision"
+	  ;;
+
 	sunos)
 	  major=".$current"
 	  versuffix=".$current.$revision"
