diff -ru Qt4-4.8.2-original/configure Qt4--current/configure
--- Qt4-4.8.2-original/configure	2012-04-26 19:46:24.000000000 +0000
+++ Qt4--current/configure	2012-06-25 02:05:49.000000000 +0000
@@ -2778,6 +2778,9 @@
 		CFG_EMBEDDED=x86
 	fi
 	;;
+    syllable:*)
+        [ -z "$PLATFORM" ] && PLATFORM=qws/linux-x86-g++
+        ;;
     *)
         echo "Qt for Embedded Linux is not supported on this platform. Disabling."
         CFG_EMBEDDED=no
diff -ru Qt4-4.8.2-original/examples/qws/qws.pro Qt4--current/examples/qws/qws.pro
--- Qt4-4.8.2-original/examples/qws/qws.pro	2012-04-26 19:45:53.000000000 +0000
+++ Qt4--current/examples/qws/qws.pro	2012-06-25 23:59:10.000000000 +0000
@@ -1,6 +1,6 @@
 TEMPLATE      = subdirs
 # no /dev/fbX
-!qnx:!vxworks:SUBDIRS = framebuffer
+#!qnx:!vxworks:SUBDIRS = framebuffer
 SUBDIRS      += mousecalibration simpledecoration
 
 # install
diff -ru Qt4-4.8.2-original/src/3rdparty/clucene/src/CLucene/config/compiler.h Qt4--current/src/3rdparty/clucene/src/CLucene/config/compiler.h
--- Qt4-4.8.2-original/src/3rdparty/clucene/src/CLucene/config/compiler.h	2012-04-26 19:46:23.000000000 +0000
+++ Qt4--current/src/3rdparty/clucene/src/CLucene/config/compiler.h	2012-06-24 04:14:05.000000000 +0000
@@ -65,7 +65,7 @@
 ////////////////////////////////////////////////////////////////////
 
 /* Operating system recognition (basic) */
-#if defined (__unix) || defined (__linux__) || defined (__QNX__) || defined (_AIX) || defined (__NetBSD__) || defined (__OpenBSD__) || defined (__Lynx__) || defined(hpux) || defined(__hpux)
+#if defined (__unix) || defined (__linux__) || defined (__QNX__) || defined (_AIX) || defined (__NetBSD__) || defined (__OpenBSD__) || defined (__Lynx__) || defined(hpux) || defined(__hpux) || defined (__SYLLABLE__)
 #undef  _UNIX
 #define _UNIX 1
 #include "CLucene/config/PlatformUnix.h"
diff -ru Qt4-4.8.2-original/src/corelib/global/qglobal.h Qt4--current/src/corelib/global/qglobal.h
--- Qt4-4.8.2-original/src/corelib/global/qglobal.h	2012-04-26 19:46:08.000000000 +0000
+++ Qt4--current/src/corelib/global/qglobal.h	2012-05-31 18:08:01.000000000 +0000
@@ -261,6 +261,8 @@
 #  define Q_OS_INTEGRITY
 #elif defined(VXWORKS) /* there is no "real" VxWorks define - this has to be set in the mkspec! */
 #  define Q_OS_VXWORKS
+#elif defined(__SYLLABLE__)
+#  define Q_OS_UNIX
 #elif defined(__MAKEDEPEND__)
 #else
 #  error "Qt has not been ported to this OS - talk to qt-bugs@trolltech.com"
diff -ru Qt4-4.8.2-original/src/corelib/io/io.pri Qt4--current/src/corelib/io/io.pri
--- Qt4-4.8.2-original/src/corelib/io/io.pri	2012-04-26 19:46:08.000000000 +0000
+++ Qt4--current/src/corelib/io/io.pri	2012-06-26 18:53:54.000000000 +0000
@@ -94,12 +94,12 @@
 
         linux-*:!symbian {
             SOURCES += \
-                    io/qfilesystemwatcher_inotify.cpp \
-                    io/qfilesystemwatcher_dnotify.cpp
+#                    io/qfilesystemwatcher_inotify.cpp \
+#                    io/qfilesystemwatcher_dnotify.cpp
 
             HEADERS += \
-                    io/qfilesystemwatcher_inotify_p.h \
-                    io/qfilesystemwatcher_dnotify_p.h
+#                    io/qfilesystemwatcher_inotify_p.h \
+#                    io/qfilesystemwatcher_dnotify_p.h
         }
 
         !nacl {
