diff -Naur scummvm-0.11.1/backends/fs/posix/posix-fs.cpp scummvm-0.11.1_old/backends/fs/posix/posix-fs.cpp
--- scummvm-0.11.1/backends/fs/posix/posix-fs.cpp	2008-02-23 22:48:21.000000000 +0000
+++ scummvm-0.11.1_old/backends/fs/posix/posix-fs.cpp	2008-03-31 22:58:57.000000000 +0000
@@ -29,6 +29,9 @@
 #ifdef MACOSX
 #include <sys/types.h>
 #endif
+#ifdef __SYLLABLE__
+#include <posix/limits.h>
+#endif
 #include <sys/param.h>
 #include <sys/stat.h>
 #include <dirent.h>
