diff -ru ffmpeg-6869-original/libavformat/allformats.c ffmpeg--current/libavformat/allformats.c
--- ffmpeg-6869-original/libavformat/allformats.c	2006-11-02 15:32:01.000000000 +0100
+++ ffmpeg--current/libavformat/allformats.c	2006-11-02 16:04:52.000000000 +0100
@@ -266,10 +266,10 @@
 #ifdef CONFIG_NUV_DEMUXER
     av_register_input_format(&nuv_demuxer);
 #endif
+#ifdef CONFIG_LIBOGG
 #ifdef CONFIG_OGG_DEMUXER
     av_register_input_format(&ogg_demuxer);
 #endif
-#ifdef CONFIG_LIBOGG
 #ifdef CONFIG_OGG_MUXER
     av_register_output_format(&ogg_muxer);
 #endif
