diff -Naur libtool-2.2.6-orig/libltdl/m4/libtool.m4 libtool-2.2.6/libltdl/m4/libtool.m4
--- libtool-2.2.6-orig/libltdl/m4/libtool.m4 2009-03-23 23:07:43.000000000 +0000
+++ libtool-2.2.6/libltdl/m4/libtool.m4 2009-03-24 01:09:19.000000000 +0000
@@ -2585,6 +2585,17 @@
postinstall_cmds='chmod +x $lib'
;;

+syllable*)
+ library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
+ soname_spec='${libname}${release}${shared_ext}$major'
+ shlibpath_var=DLL_PATH
+ shlibpath_overrides_runpath=yes
+ version_type=linux
+ dynamic_linker="Syllable"
+ need_lib_prefix=no
+ need_version=no
+ ;;
+
sunos4*)
version_type=sunos
library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${shared_ext}$versuffix'
@@ -3114,6 +3125,10 @@
lt_cv_deplibs_check_method=pass_all
;;

+syllable*)
+ lt_cv_deplibs_check_method=pass_all
+ ;;
+
sysv5* | sco3.2v5* | sco5v6* | unixware* | OpenUNIX* | sysv4*uw2*)
lt_cv_deplibs_check_method=pass_all
;;
diff -Naur libtool-2.2.6-orig/libltdl/m4/ltdl.m4 libtool-2.2.6/libltdl/m4/ltdl.m4
--- libtool-2.2.6-orig/libltdl/m4/ltdl.m4 2009-03-23 23:07:43.000000000 +0000
+++ libtool-2.2.6/libltdl/m4/ltdl.m4 2009-03-24 01:10:00.000000000 +0000
@@ -521,6 +521,9 @@
solaris*)
lt_cv_sys_dlopen_deplibs=yes
;;
+ syllable*)
+ lt_cv_sys_dlopen_deplibs=yes
+ ;;
sysv5* | sco3.2v5* | sco5v6* | unixware* | OpenUNIX* | sysv4*uw2*)
libltdl_cv_sys_dlopen_deplibs=yes
;;
