diff -ru linux-2.6.21.1-original/Makefile linux-2.6.21.1/Makefile
--- linux-2.6.21.1-original/Makefile	2007-04-27 21:49:26.000000000 +0000
+++ linux-2.6.21.1/Makefile	2007-07-03 11:40:18.000000000 +0000
@@ -988,7 +988,7 @@
 ifeq "$(strip $(INSTALL_MOD_PATH))" ""
 depmod_opts	:=
 else
-depmod_opts	:= -b $(INSTALL_MOD_PATH) -r
+depmod_opts	:= -b $(IMAGE) -r
 endif
 PHONY += _modinst_post
 _modinst_post: _modinst_
@@ -992,7 +992,7 @@
 endif
 PHONY += _modinst_post
 _modinst_post: _modinst_
-	if [ -r System.map -a -x $(DEPMOD) ]; then $(DEPMOD) -ae -F System.map $(depmod_opts) $(KERNELRELEASE); fi
+	if [ -r System.map -a -x $(DEPMOD) ]; then depmod -ae -F System.map $(depmod_opts) $(KERNELRELEASE); fi
 
 else # CONFIG_MODULES
 
