diff -urN Transmission-1.72_orig/third-party/libnatpmp/getgateway.c Transmission-1.72/third-party/libnatpmp/getgateway.c
--- Transmission-1.72_orig/third-party/libnatpmp/getgateway.c	2009-06-17 00:01:54.000000000 +0000
+++ Transmission-1.72/third-party/libnatpmp/getgateway.c	2009-07-11 12:39:24.000000000 +0000
@@ -421,3 +421,10 @@
 }
 #endif /* #ifdef USE_WIN32_CODE */
 
+#ifdef __SYLLABLE__
+int getdefaultgateway(in_addr_t *addr)
+{
+  /* Not implemented */
+  return 0;
+}
+#endif
