diff -ru unzip-5.51.original/acorn/acorn.c unzip-5.51/acorn/acorn.c
--- unzip-5.51.original/acorn/acorn.c	2003-09-26 22:30:31.000000000 +0000
+++ unzip-5.51/acorn/acorn.c	2004-10-19 01:56:05.000000000 +0000
@@ -190,6 +190,7 @@
         case VMS_:
         case ATARI_:
         case BEOS_:
+        case ATHEOS_:
         case QDOS_:
         case TANDEM_:
             G.pInfo->file_attr = (unsigned)(tmp >> 16);
diff -ru unzip-5.51.original/amiga/amiga.c unzip-5.51/amiga/amiga.c
--- unzip-5.51.original/amiga/amiga.c	2003-09-26 22:31:29.000000000 +0000
+++ unzip-5.51/amiga/amiga.c	2004-10-19 01:56:05.000000000 +0000
@@ -218,6 +218,7 @@
         case ACORN_:
         case ATARI_:
         case BEOS_:
+        case ATHEOS_:
         case QDOS_:
         case TANDEM_:
             {
diff -ru unzip-5.51.original/aosvs/aosvs.c unzip-5.51/aosvs/aosvs.c
--- unzip-5.51.original/aosvs/aosvs.c	2004-04-12 00:29:57.000000000 +0000
+++ unzip-5.51/aosvs/aosvs.c	2004-10-19 01:56:05.000000000 +0000
@@ -399,6 +399,7 @@
         case ACORN_:
         case ATARI_:
         case BEOS_:
+        case ATHEOS_:
         case QDOS_:
         case TANDEM_:
             G.pInfo->file_attr = (unsigned)(tmp >> 16);
diff -ru unzip-5.51.original/atari/atari.c unzip-5.51/atari/atari.c
--- unzip-5.51.original/atari/atari.c	2004-04-12 00:22:15.000000000 +0000
+++ unzip-5.51/atari/atari.c	2004-10-19 01:56:05.000000000 +0000
@@ -188,6 +188,7 @@
         case ACORN_:
         case ATARI_:
         case BEOS_:
+        case ATHEOS_:
         case QDOS_:
         case TANDEM_:
             G.pInfo->file_attr = (unsigned)(tmp >> 16);
diff -ru unzip-5.51.original/beos/beos.c unzip-5.51/beos/beos.c
--- unzip-5.51.original/beos/beos.c	2004-04-12 00:26:15.000000000 +0000
+++ unzip-5.51/beos/beos.c	2004-10-19 01:56:05.000000000 +0000
@@ -225,6 +225,7 @@
                 tmp &= 0x41FFFFFFL;     /* leave directory bit as set */
             /* fall through! */
         case BEOS_:
+        case ATHEOS_:
         case UNIX_:
         case VMS_:
         case ACORN_:
diff -ru unzip-5.51.original/extract.c unzip-5.51/extract.c
--- unzip-5.51.original/extract.c	2004-04-16 00:58:01.000000000 +0000
+++ unzip-5.51/extract.c	2004-10-19 01:56:05.000000000 +0000
@@ -1435,7 +1435,7 @@
      * to disk, prepare to restore the link */
     if (S_ISLNK(G.pInfo->file_attr) &&
         (G.pInfo->hostnum == UNIX_ || G.pInfo->hostnum == ATARI_ ||
-         G.pInfo->hostnum == BEOS_) &&
+         G.pInfo->hostnum == BEOS_ || G.pInfo->hostnum == ATHEOS_) &&
         !uO.tflag && !uO.cflag && (G.lrec.ucsize > 0))
         G.symlnk = TRUE;
     else
@@ -1777,6 +1777,7 @@
             case EF_ACL:
             case EF_MAC3:
             case EF_BEOS:
+            case EF_ATHEOS:
                 switch (ebID) {
                   case EF_OS2:
                   case EF_ACL:
@@ -1799,6 +1800,14 @@
                     else
                         eb_cmpr_offs = EB_BEOS_HLEN;
                     break;
+                  case EF_ATHEOS:
+                    if (ebLen >= EB_ATHEOS_HLEN &&
+                        (*(ef+(EB_HEADSIZE+EB_FLGS_OFFS)) & EB_AT_FL_UNCMPR) &&
+                        (makelong(ef+EB_HEADSIZE) == ebLen - EB_ATHEOS_HLEN))
+                        eb_cmpr_offs = 0;
+                    else
+                        eb_cmpr_offs = EB_ATHEOS_HLEN;
+                  break;
                 }
                 if ((r = test_compr_eb(__G__ ef, ebLen, eb_cmpr_offs, NULL))
                     != PK_OK) {
diff -ru unzip-5.51.original/human68k/human68k.c unzip-5.51/human68k/human68k.c
--- unzip-5.51.original/human68k/human68k.c	2003-09-26 22:34:55.000000000 +0000
+++ unzip-5.51/human68k/human68k.c	2004-10-19 01:56:05.000000000 +0000
@@ -193,6 +193,7 @@
         case ACORN_:
         case ATARI_:
         case BEOS_:
+        case ATHEOS_:
         case QDOS_:
             G.pInfo->file_attr = _mode2dos(tmp >> 16);
             return 0;
diff -ru unzip-5.51.original/novell/novell.c unzip-5.51/novell/novell.c
--- unzip-5.51.original/novell/novell.c	2003-09-26 22:37:19.000000000 +0000
+++ unzip-5.51/novell/novell.c	2004-10-19 01:56:05.000000000 +0000
@@ -218,6 +218,7 @@
         case ACORN_:
         case ATARI_:
         case BEOS_:
+        case ATHEOS_:
         case QDOS_:
         case TANDEM_:
             G.pInfo->file_attr = (unsigned)(tmp >> 16);
diff -ru unzip-5.51.original/process.c unzip-5.51/process.c
--- unzip-5.51.original/process.c	2003-10-19 22:52:05.000000000 +0000
+++ unzip-5.51/process.c	2004-10-19 01:56:05.000000000 +0000
@@ -1155,7 +1155,7 @@
                 break;
 
             default:     /* AMIGA_, FS_HPFS_, FS_NTFS_, MAC_, UNIX_, ATARI_, */
-                break;   /*  FS_VFAT_, BEOS_ (Z_SYSTEM_), THEOS_: */
+                break;   /*  FS_VFAT_, ATHEOS_, BEOS_ (Z_SYSTEM_), THEOS_: */
                          /*  no conversion */
         }
     else if (uO.L_flag > 1)   /* let -LL force lower case for all names */
diff -ru unzip-5.51.original/qdos/qdos.c unzip-5.51/qdos/qdos.c
--- unzip-5.51.original/qdos/qdos.c	2004-05-23 01:02:15.000000000 +0000
+++ unzip-5.51/qdos/qdos.c	2004-10-19 01:56:05.000000000 +0000
@@ -368,6 +368,7 @@
         case ACORN_:
         case ATARI_:
         case BEOS_:
+        case ATHEOS_:
         case TANDEM_:
             G.pInfo->file_attr = (unsigned)(tmp >> 16);
             if (G.pInfo->file_attr != 0 || !G.extra_field) {
diff -ru unzip-5.51.original/tandem/tanunz.c unzip-5.51/tandem/tanunz.c
--- unzip-5.51.original/tandem/tanunz.c	2004-05-23 01:02:23.000000000 +0000
+++ unzip-5.51/tandem/tanunz.c	2004-10-19 01:56:05.000000000 +0000
@@ -368,6 +368,7 @@
         case ACORN_:
         case ATARI_:
         case BEOS_:
+        case ATHEOS_:
         case QDOS_:
             G.pInfo->file_attr = (unsigned)(tmp >> 16);
             if (G.pInfo->file_attr != 0 || !G.extra_field) {
diff -ru unzip-5.51.original/theos/theos.c unzip-5.51/theos/theos.c
--- unzip-5.51.original/theos/theos.c	2004-05-23 01:02:20.000000000 +0000
+++ unzip-5.51/theos/theos.c	2004-10-19 01:56:05.000000000 +0000
@@ -294,6 +294,7 @@
         case ACORN_:
         case ATARI_:
         case BEOS_:
+        case ATHEOS_:
         case QDOS_:
         case TANDEM_:
         case THEOS_:
diff -ru unzip-5.51.original/tops20/tops20.c unzip-5.51/tops20/tops20.c
--- unzip-5.51.original/tops20/tops20.c	2000-04-15 00:20:21.000000000 +0000
+++ unzip-5.51/tops20/tops20.c	2004-10-19 01:56:05.000000000 +0000
@@ -44,6 +44,7 @@
         case ACORN_:
         case ATARI_:
         case BEOS_:
+        case ATHEOS_:
         case QDOS_:
             G.pInfo->file_attr = (unsigned)(tmp >> 16);
             break;
diff -ru unzip-5.51.original/ttyio.c unzip-5.51/ttyio.c
--- unzip-5.51.original/ttyio.c	2001-02-12 23:18:23.000000000 +0000
+++ unzip-5.51/ttyio.c	2004-10-19 01:56:05.000000000 +0000
@@ -54,6 +54,10 @@
 #  define HAVE_TERMIOS_H
 #endif
 
+#ifdef __SYLLABLE__
+#  define HAVE_TERMIOS_H
+#endif
+
 #ifdef _POSIX_VERSION
 #  ifndef USE_POSIX_TERMIOS
 #    define USE_POSIX_TERMIOS  /* use POSIX style termio (termios) */
@@ -332,7 +336,7 @@
 
 #if (defined(UNZIP) && !defined(FUNZIP))
 
-#if (defined(UNIX) || defined(__BEOS__))
+#if (defined(UNIX) || defined(__BEOS__) || defined(__SYLLABLE__))
 #ifdef MORE
 
 /*
@@ -472,7 +476,7 @@
 }
 
 
-#else /* !UNIX && !__BEOS__ */
+#else /* !UNIX && !__BEOS__ && !__SYLLABLE__ */
 #ifndef VMS     /* VMS supplies its own variant of getch() */
 
 
@@ -584,7 +588,7 @@
 #else /* !HAVE_WORKING_GETCH */
 
 
-#if (defined(UNIX) || defined(__MINT__) || defined(__BEOS__))
+#if (defined(UNIX) || defined(__MINT__) || defined(__BEOS__) || defined(__SYLLABLE__))
 
 #ifndef _PATH_TTY
 #  ifdef __MINT__
@@ -641,7 +645,7 @@
 
 } /* end function getp() */
 
-#endif /* UNIX || __MINT__ || __BEOS__ */
+#endif /* UNIX || __MINT__ || __BEOS__ || __SYLLABLE__ */
 
 
 
diff -ru unzip-5.51.original/unix/unix.c unzip-5.51/unix/unix.c
--- unzip-5.51.original/unix/unix.c	2004-04-12 00:29:59.000000000 +0000
+++ unzip-5.51/unix/unix.c	2004-10-19 01:56:05.000000000 +0000
@@ -316,6 +316,7 @@
         case ACORN_:
         case ATARI_:
         case BEOS_:
+        case ATHEOS_:
         case QDOS_:
         case TANDEM_:
             G.pInfo->file_attr = (unsigned)(tmp >> 16);
diff -ru unzip-5.51.original/unzip.c unzip-5.51/unzip.c
--- unzip-5.51.original/unzip.c	2004-05-23 01:01:13.000000000 +0000
+++ unzip-5.51/unzip.c	2004-10-19 01:56:05.000000000 +0000
@@ -49,8 +49,8 @@
 
   Version:  unzip5??.{tar.Z | tar.gz | zip} for Unix, VMS, OS/2, MS-DOS, Amiga,
               Atari, Windows 3.x/95/NT/CE, Macintosh, Human68K, Acorn RISC OS,
-              BeOS, SMS/QDOS, VM/CMS, MVS, AOS/VS, Tandem NSK, Theos and
-              TOPS-20.
+              BeOS, AtheOS, SMS/QDOS, VM/CMS, MVS, AOS/VS, Tandem NSK, Theos
+              and TOPS-20.
 
   Copyrights:  see accompanying file "LICENSE" in UnZip source distribution.
                (This software is free but NOT IN THE PUBLIC DOMAIN.)
@@ -1254,15 +1254,15 @@
                     else
                         uO.jflag = TRUE;
                     break;
-#if (defined(__BEOS__) || defined(MACOS))
-                case ('J'):    /* Junk BeOS or MacOS file attributes */
+#if (defined(__BEOS__) || defined(MACOS) || defined(__SYLLABLE__))
+                case ('J'):    /* Junk AtheOS, BeOS or MacOS file attributes */
                     if( negative ) {
                         uO.J_flag = FALSE, negative = 0;
                     } else {
                         uO.J_flag = TRUE;
                     }
                     break;
-#endif /* __BEOS__ || MACOS */
+#endif /* __BEOS__ || MACOS || __SYLLABLE__ */
 #ifndef SFX
                 case ('l'):
                     if (negative) {
@@ -2005,3 +2005,4 @@
 
 #endif /* !SFX */
 #endif /* !WINDLL */
+ 
diff -ru unzip-5.51.original/unzip.h unzip-5.51/unzip.h
--- unzip-5.51.original/unzip.h	2004-05-23 01:04:24.000000000 +0000
+++ unzip-5.51/unzip.h	2004-10-19 01:56:05.000000000 +0000
@@ -243,7 +243,7 @@
 #    define MODERN
 #  endif
 #endif
-#if (defined(CMS_MVS) || defined(__BEOS__))  /* || defined(CONVEX) */
+#if (defined(CMS_MVS) || defined(__BEOS__) || defined(__SYLLABLE__))  /* || defined(CONVEX) */
 #  ifndef PROTO
 #    define PROTO
 #  endif
@@ -450,8 +450,8 @@
     int scanimage;      /* -I: scan image files */
 #endif
     int jflag;          /* -j: junk pathnames (unzip) */
-#if (defined(__BEOS__) || defined(MACOS))
-    int J_flag;         /* -J: ignore BeOS/MacOS extra field info (unzip) */
+#if (defined(__BEOS__) || defined(MACOS) || defined(__SYLLABLE__))
+    int J_flag;         /* -J: ignore AtheOS/BeOS/MacOS extra field info (unzip) */
 #endif
     int lflag;          /* -12slmv: listing format (zipinfo) */
     int L_flag;         /* -L: convert filenames from some OSes to lowercase */
@@ -480,7 +480,7 @@
     int uflag;          /* -u: "update" (extract only newer/brand-new files) */
     int vflag;          /* -v: (verbosely) list directory */
     int V_flag;         /* -V: don't strip VMS version numbers */
-#if (defined(__BEOS__) || defined(TANDEM) || defined(THEOS) || defined(UNIX))
+#if (defined(__BEOS__) || defined(TANDEM) || defined(THEOS) || defined(UNIX) || defined (__SYLLABLE__))
     int X_flag;         /* -X: restore owner/protection or UID/GID or ACLs */
 #else
 #if (defined(OS2) || defined(VMS) || defined(WIN32))
diff -ru unzip-5.51.original/unzpriv.h unzip-5.51/unzpriv.h
--- unzip-5.51.original/unzpriv.h	2004-03-01 18:09:05.000000000 +0000
+++ unzip-5.51/unzpriv.h	2004-10-19 01:56:05.000000000 +0000
@@ -310,6 +310,43 @@
 #endif
 
 /*---------------------------------------------------------------------------
+    AtheOS section:
+  ---------------------------------------------------------------------------*/
+
+#ifdef __SYLLABLE__
+#  include <sys/types.h>
+#  include <sys/stat.h>
+#  include <sys/param.h>
+#  include <fcntl.h>
+#  include <limits.h>
+#  include <time.h>
+#  include <unistd.h>
+#  include <utime.h>
+
+#  define GOT_UTIMBUF
+#  define NO_STRNICMP
+#  define DIRENT
+#  ifndef DATE_FORMAT
+#    define DATE_FORMAT DF_MDY  /* GRR:  customize with locale.h somehow? */
+#  endif
+#  define lenEOL        1
+#  define PutNativeEOL  *q++ = native(LF);
+#  define SCREENSIZE(ttrows, ttcols)  screensize(ttrows, ttcols)
+#  define SCREENWIDTH 80
+#  if (!defined(NO_EF_UT_TIME) && !defined(USE_EF_UT_TIME))
+#    define USE_EF_UT_TIME
+#  endif
+#  define SET_SYMLINK_ATTRIBS
+#  define SET_DIR_ATTRIB
+#  if (!defined(NOTIMESTAMP) && !defined(TIMESTAMP))
+#    define TIMESTAMP
+#  endif
+#  define RESTORE_UIDGID
+#  define INT_SPRINTF
+#  define SYMLINKS
+#endif
+
+/*---------------------------------------------------------------------------
     BeOS section:
   ---------------------------------------------------------------------------*/
 
@@ -739,7 +776,7 @@
 #  define DOS_T20_VMS
 #endif
 
-#if (defined(__BEOS__) || defined(UNIX))
+#if (defined(__BEOS__) || defined(UNIX) || defined(__SYLLABLE__))
 #  define BEO_UNX
 #endif
 
@@ -1378,7 +1415,8 @@
 #define BEOS_             16   /* hybrid POSIX/database filesystem */
 #define TANDEM_           17   /* Tandem NSK */
 #define THEOS_            18   /* THEOS */
-#define NUM_HOSTS         19   /* index of last system + 1 */
+#define ATHEOS_           30   /* AtheOS */
+#define NUM_HOSTS         31   /* index of last system + 1 */
 
 #define STORED            0    /* compression methods */
 #define SHRUNK            1
@@ -1425,6 +1463,7 @@
 #define EF_ACL       0x4c41    /* (OS/2) access control list ("AL") */
 #define EF_NTSD      0x4453    /* NT security descriptor ("SD") */
 #define EF_BEOS      0x6542    /* BeOS ("Be") */
+#define EF_ATHEOS    0x7441    /* AtheOS ("At") */
 #define EF_QDOS      0xfb4a    /* SMS/QDOS ("J\373") */
 #define EF_AOSVS     0x5356    /* AOS/VS ("VS") */
 #define EF_SPARK     0x4341    /* David Pilling's Acorn/SparkFS ("AC") */
@@ -1460,10 +1499,12 @@
 #define EB_UT_FL_CTIME    (1 << 2)      /* ctime present */
 
 #define EB_FLGS_OFFS      4    /* offset of flags area in generic compressed
-                                  extra field blocks (BEOS, MAC, and others) */
+                                  extra field blocks (BEOS, ATHEOS, MAC, and others) */
 #define EB_OS2_HLEN       4    /* size of OS2/ACL compressed data header */
 #define EB_BEOS_HLEN      5    /* length of BeOS e.f attribute header */
+#define EB_ATHEOS_HLEN    5    /* length of AtheOS e.f. attribute header */
 #define EB_BE_FL_UNCMPR   0x01 /* "BeOS attributes uncompressed" bit flag */
+#define EB_AT_FL_UNCMPR   0x01 /* "AtheOS attributes uncompressed" bit flag */
 #define EB_MAC3_HLEN      14   /* length of Mac3 attribute block header */
 #define EB_SMARTZIP_HLEN  64   /* fixed length of the SmartZip extra field */
 #define EB_M3_FL_DATFRK   0x01 /* "this entry is data fork" flag */
@@ -2142,7 +2183,7 @@
    void  close_outfile   OF((__GPRO));                              /* local */
 #endif
 #ifdef SET_SYMLINK_ATTRIBS
-   int  set_symlnk_attribs  OF((__GPRO__ slinkentry *slnk_entry))   /* local */
+   int  set_symlnk_attribs  OF((__GPRO__ slinkentry *slnk_entry));   /* local */
 #endif
 #ifdef SET_DIR_ATTRIB
    int   defer_dir_attribs  OF((__GPRO__ direntry **pd));           /* local */
diff -ru unzip-5.51.original/vms/vms.c unzip-5.51/vms/vms.c
--- unzip-5.51.original/vms/vms.c	2003-09-26 22:57:01.000000000 +0000
+++ unzip-5.51/vms/vms.c	2004-10-19 01:56:05.000000000 +0000
@@ -2321,6 +2321,7 @@
         case ACORN_:
         case ATARI_:
         case BEOS_:
+        case ATHEOS_:
         case QDOS_:
         case TANDEM_:
             {
diff -ru unzip-5.51.original/win32/win32.c unzip-5.51/win32/win32.c
--- unzip-5.51.original/win32/win32.c	2004-03-01 18:55:25.000000000 +0000
+++ unzip-5.51/win32/win32.c	2004-10-19 01:56:05.000000000 +0000
@@ -460,6 +460,7 @@
             case EF_MVS:
             case EF_ACL:
             case EF_BEOS:
+            case EF_ATHEOS:
             case EF_QDOS:
             case EF_AOSVS:
             case EF_SPARK:
diff -ru unzip-5.51.original/zipinfo.c unzip-5.51/zipinfo.c
--- unzip-5.51.original/zipinfo.c	2002-12-16 22:02:33.000000000 +0000
+++ unzip-5.51/zipinfo.c	2004-10-19 01:56:05.000000000 +0000
@@ -192,6 +192,7 @@
 static ZCONST char Far OS_MVS[] = "MVS";
 static ZCONST char Far OS_VFAT[] = "Win32 VFAT";
 static ZCONST char Far OS_BeOS[] = "BeOS";
+static ZCONST char Far OS_AtheOS[] = "AtheOS";
 static ZCONST char Far OS_Tandem[] = "Tandem NSK";
 static ZCONST char Far OS_Theos[] = "Theos";
 #ifdef OLD_THEOS_EXTRA
@@ -323,6 +324,7 @@
 static ZCONST char Far efACL[] = "OS/2 ACL";
 static ZCONST char Far efNTSD[] = "Security Descriptor";
 static ZCONST char Far efBeOS[] = "BeOS";
+static ZCONST char Far efAtheOS[] = "AtheOS";
 static ZCONST char Far efQDOS[] = "SMS/QDOS";
 static ZCONST char Far efAOSVS[] = "AOS/VS";
 static ZCONST char Far efSpark[] = "Acorn SparkFS";
@@ -370,6 +372,8 @@
 static ZCONST char Far BeOSdata[] = ".\n\
     The local extra field has %lu bytes of %scompressed BeOS file attributes";
  /* The associated file has type code `%c%c%c%c' and creator code `%c%c%c%c'" */
+static ZCONST char Far AtheOSdata[] = ".\n\
+    The local extra field has %lu bytes of %scompressed AtheOS file attributes";
 static ZCONST char Far QDOSdata[] = ".\n\
     The QDOS extra field subtype is `%c%c%c%c'";
 static ZCONST char Far AOSVSdata[] = ".\n\
@@ -943,7 +947,8 @@
     static ZCONST char Far *os[NUM_HOSTS] = {
         OS_FAT, OS_Amiga, OS_VMS, OS_Unix, OS_VMCMS, OS_AtariST, OS_HPFS,
         OS_Macintosh, OS_ZSystem, OS_CPM, OS_TOPS20, OS_NTFS, OS_QDOS,
-        OS_Acorn, OS_VFAT, OS_MVS, OS_BeOS, OS_Tandem, OS_Theos
+        OS_Acorn, OS_VFAT, OS_MVS, OS_BeOS, OS_Tandem, OS_Theos, "",
+		"", "", "", "", "", "", "", "", "", "", OS_AtheOS
     };
     static ZCONST char Far *method[NUM_METHODS] = {
         MthdNone, MthdShrunk, MthdRedF1, MthdRedF2, MthdRedF3, MthdRedF4,
@@ -1395,6 +1400,9 @@
                 case EF_BEOS:
                     ef_fieldname = efBeOS;
                     break;
+                case EF_ATHEOS:
+                    ef_fieldname = efAtheOS;
+                    break;
                 case EF_QDOS:
                     ef_fieldname = efQDOS;
                     break;
@@ -1633,6 +1641,22 @@
                         }
                     }
                     break;
+
+                case EF_ATHEOS:
+                    if (eb_datalen >= EB_ATHEOS_HLEN) {
+                        ulg eb_uc = makelong(ef_ptr);
+                        unsigned eb_is_uc = *(ef_ptr+EB_FLGS_OFFS) & EB_AT_FL_UNCMPR;
+                        Info(slide, 0, ((char *)slide, LoadFarString(AtheOSdata),
+                          eb_uc, eb_is_uc ? "un" : nullStr));
+                        if (eb_is_uc) {
+                            if (*pEndprev > 0L)
+                                *pEndprev += makelong(ef_ptr);
+                        } else {
+                            *pEndprev = 0L; /* no clue about csize of local */
+                        }
+                    }
+                    break;
+
                 case EF_QDOS:
                     if (eb_datalen >= 4) {
                         Info(slide, 0, ((char *)slide, LoadFarString(QDOSdata),
