diff -ru sdl-1.2.13-original/acinclude.m4 sdl-1.2.13/acinclude.m4
--- sdl-1.2.13-original/acinclude.m4	2007-12-31 05:48:39.000000000 +0100
+++ sdl-1.2.13/acinclude.m4	2008-12-11 00:40:46.000000000 +0100
@@ -1245,6 +1245,12 @@
     lt_cv_dlopen_self=yes
     ;;
 
+  syllable*)
+    lt_cv_dlopen="load_library"
+    lt_cv_dlopen_libs=
+    lt_cv_dlopen_self=yes
+    ;;
+
   mingw* | pw32*)
     lt_cv_dlopen="LoadLibrary"
     lt_cv_dlopen_libs=
@@ -1609,6 +1615,12 @@
   shlibpath_var=LIBRARY_PATH
   ;;
 
+syllable*)
+  library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}${major} ${libname}${shared_ext}'
+  dynamic_linker="$host_os ld.so"
+  shlibpath_var=LIBRARY_PATH
+  ;;
+
 bsdi[[45]]*)
   version_type=linux
   need_version=no
@@ -2606,6 +2618,11 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+syllable*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
+
 bsdi[[45]]*)
   lt_cv_deplibs_check_method='file_magic ELF [[0-9]][[0-9]]*-bit [[ML]]SB (shared object|dynamic lib)'
   lt_cv_file_magic_cmd='/usr/bin/file -L'
@@ -5165,7 +5182,7 @@
       # like `-m68040'.
       _LT_AC_TAGVAR(lt_prog_compiler_pic, $1)='-m68020 -resident32 -malways-restore-a4'
       ;;
-    beos* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
+    beos* | syllable* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
       # PIC is the default for these OSes.
       ;;
     mingw* | os2* | pw32*)
@@ -5438,7 +5455,7 @@
       _LT_AC_TAGVAR(lt_prog_compiler_pic, $1)='-m68020 -resident32 -malways-restore-a4'
       ;;
 
-    beos* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
+    beos* | syllable* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
       # PIC is the default for these OSes.
       ;;
 
diff -ru sdl-1.2.13-original/configure.in sdl-1.2.13/configure.in
--- sdl-1.2.13-original/configure.in	2007-12-31 05:48:39.000000000 +0100
+++ sdl-1.2.13/configure.in	2008-12-11 00:40:47.000000000 +0100
@@ -1555,6 +1555,14 @@
     fi
 }
 
+CheckSyllable()
+{
+   AC_DEFINE(SDL_VIDEO_DRIVER_SYLLABLE)
+   SOURCES="$SOURCES $srcdir/src/video/syllable/*.cc"
+   have_video=yes
+}
+
+
 dnl Check to see if OpenGL support is desired
 AC_ARG_ENABLE(video-opengl,
 AC_HELP_STRING([--enable-video-opengl], [include OpenGL context creation [[default=yes]]]),
@@ -2507,6 +2515,48 @@
         SOURCES="$srcdir/src/main/beos/*.cc $SOURCES"
         EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lroot -lbe -lmedia -lgame -ldevice -ltextencoding"
         ;;
+    *-*-syllable*)
+        ARCH=syllable
+        ac_default_prefix=/resources/sdl
+        CheckDummyVideo
+        CheckDiskAudio
+        CheckNASM
+		CheckDLOPEN
+        CheckSyllable
+        # Set up files for the audio library
+        if test x$enable_audio = xyes; then
+            SOURCES="$SOURCES $srcdir/src/audio/syllable/*.cc"
+            have_audio=yes
+        fi
+        # Set up files for the joystick library
+        # (No joystick support yet)
+        if test x$enable_joystick = xyes; then
+            JOYSTICK_SUBDIRS="$JOYSTICK_SUBDIRS dummy"
+            JOYSTICK_DRIVERS="$JOYSTICK_DRIVERS dummy/libjoystick_dummy.la"
+        fi
+        # Set up files for the cdrom library
+        # (No cdrom support yet)
+        if test x$enable_cdrom = xyes; then
+            CDROM_SUBDIRS="$CDROM_SUBDIRS dummy"
+            CDROM_DRIVERS="$CDROM_DRIVERS dummy/libcdrom_dummy.la"
+        fi
+        # Set up files for the thread library
+        if test x$enable_threads = xyes; then
+            AC_DEFINE(SDL_THREAD_SYLLABLE)
+            SOURCES="$SOURCES $srcdir/src/thread/syllable/*.c"
+            SOURCES="$SOURCES $srcdir/src/thread/generic/SDL_sysmutex.c"
+            SOURCES="$SOURCES $srcdir/src/thread/generic/SDL_syscond.c"
+            have_threads=yes
+        fi
+        # Set up files for the timer library
+        if test x$enable_timers = xyes; then
+            AC_DEFINE(SDL_TIMER_SYLLABLE)
+            SOURCES="$SOURCES $srcdir/src/timer/syllable/*.c"
+            have_timers=yes
+        fi
+        # The Syllable platform requires special libraries
+        SYSTEM_LIBS="$SYSTEM_LIBS -lsyllable -lmedia"
+        ;;
     *-*-darwin* )
         # This could be either full "Mac OS X", or plain "Darwin" which is
         # just the OS X kernel sans upper layers like Carbon and Cocoa.
diff -ru sdl-1.2.13-original/include/SDL_platform.h sdl-1.2.13/include/SDL_platform.h
--- sdl-1.2.13-original/include/SDL_platform.h	2007-12-31 05:48:36.000000000 +0100
+++ sdl-1.2.13/include/SDL_platform.h	2008-12-11 00:40:47.000000000 +0100
@@ -29,6 +29,10 @@
 #undef __AIX__
 #define __AIX__		1
 #endif
+#if defined(__SYLLABLE__)
+#undef __SYLLABLE__
+#define __SYLLABLE__  1
+#endif
 #if defined(__BEOS__)
 #undef __BEOS__
 #define __BEOS__	1
diff -ru sdl-1.2.13-original/src/audio/SDL_audio.c sdl-1.2.13/src/audio/SDL_audio.c
--- sdl-1.2.13-original/src/audio/SDL_audio.c	2007-12-31 05:47:59.000000000 +0100
+++ sdl-1.2.13/src/audio/SDL_audio.c	2008-12-11 00:40:47.000000000 +0100
@@ -76,6 +76,9 @@
 #if SDL_AUDIO_DRIVER_PAUD
 	&Paud_bootstrap,
 #endif
+#ifdef __SYLLABLE__
+	&SYLLABLE_Audio_bootstrap,
+#endif
 #if SDL_AUDIO_DRIVER_BAUDIO
 	&BAUDIO_bootstrap,
 #endif
diff -ru sdl-1.2.13-original/src/audio/SDL_sysaudio.h sdl-1.2.13/src/audio/SDL_sysaudio.h
--- sdl-1.2.13-original/src/audio/SDL_sysaudio.h	2007-12-31 05:47:59.000000000 +0100
+++ sdl-1.2.13/src/audio/SDL_sysaudio.h	2008-12-11 00:40:47.000000000 +0100
@@ -140,6 +140,9 @@
 #if SDL_AUDIO_DRIVER_PAUD
 extern AudioBootStrap Paud_bootstrap;
 #endif
+#ifdef __SYLLABLE__
+extern AudioBootStrap SYLLABLE_Audio_bootstrap;
+#endif
 #if SDL_AUDIO_DRIVER_BAUDIO
 extern AudioBootStrap BAUDIO_bootstrap;
 #endif
diff -ru sdl-1.2.13-original/src/thread/SDL_thread_c.h sdl-1.2.13/src/thread/SDL_thread_c.h
--- sdl-1.2.13-original/src/thread/SDL_thread_c.h	2007-12-31 05:48:16.000000000 +0100
+++ sdl-1.2.13/src/thread/SDL_thread_c.h	2008-12-11 00:40:47.000000000 +0100
@@ -43,6 +43,8 @@
 #include "win32/SDL_systhread_c.h"
 #elif SDL_THREAD_SYMBIAN
 #include "symbian/SDL_systhread_c.h"
+#elif __SYLLABLE__
+#include "syllable/SDL_systhread_c.h"
 #else
 #error Need thread implementation for this platform
 #include "generic/SDL_systhread_c.h"
diff -ru sdl-1.2.13-original/src/video/SDL_sysvideo.h sdl-1.2.13/src/video/SDL_sysvideo.h
--- sdl-1.2.13-original/src/video/SDL_sysvideo.h	2007-12-31 05:48:14.000000000 +0100
+++ sdl-1.2.13/src/video/SDL_sysvideo.h	2008-12-11 00:40:47.000000000 +0100
@@ -368,6 +368,9 @@
 #if SDL_VIDEO_DRIVER_BWINDOW
 extern VideoBootStrap BWINDOW_bootstrap;
 #endif
+#ifdef __SYLLABLE__
+extern VideoBootStrap SYLLABLE_Video_bootstrap;
+#endif
 #if SDL_VIDEO_DRIVER_TOOLBOX
 extern VideoBootStrap TOOLBOX_bootstrap;
 #endif
diff -ru sdl-1.2.13-original/src/video/SDL_video.c sdl-1.2.13/src/video/SDL_video.c
--- sdl-1.2.13-original/src/video/SDL_video.c	2007-12-31 05:48:14.000000000 +0100
+++ sdl-1.2.13/src/video/SDL_video.c	2008-12-11 00:40:47.000000000 +0100
@@ -84,6 +84,9 @@
 #if SDL_VIDEO_DRIVER_BWINDOW
 	&BWINDOW_bootstrap,
 #endif
+#ifdef __SYLLABLE__
+	&SYLLABLE_Video_bootstrap,
+#endif
 #if SDL_VIDEO_DRIVER_TOOLBOX
 	&TOOLBOX_bootstrap,
 #endif
