diff -ru samba-3.0.2a.orig/source/include/config.h.in samba-3.0.2a/source/include/config.h.in
--- samba-3.0.2a.orig/source/include/config.h.in	2004-02-13 19:57:28.000000000 +0000
+++ samba-3.0.2a/source/include/config.h.in	2004-02-21 03:18:06.000000000 +0000
@@ -303,7 +303,7 @@
 #undef HAVE_FTELLO64
 
 /* Define to 1 if you have the `ftruncate' function. */
-#undef HAVE_FTRUNCATE
+#define HAVE_FTRUNCATE 1
 
 /* Define to 1 if you have the `ftruncate64' function. */
 #undef HAVE_FTRUNCATE64
