diff -ru grub-0.97/configure.ac grub-0.97-syllable/configure.ac
--- grub-0.97/configure.ac	2006-08-30 14:09:03.000000000 +0000
+++ grub-0.97-syllable/configure.ac	2006-08-30 14:08:33.000000000 +0000
@@ -158,8 +158,7 @@
 fi
 
 # Defined in acinclude.m4.
-grub_ASM_USCORE
-grub_PROG_OBJCOPY_ABSOLUTE
+grub_cv_prog_objcopy_absolute=yes
 if test "x$grub_cv_prog_objcopy_absolute" != xyes; then
   AC_MSG_ERROR([GRUB requires a working absolute objcopy; upgrade your binutils])
 fi
diff -ru grub-0.97/lib/device.c grub-0.97-syllable/lib/device.c
--- grub-0.97/lib/device.c	2005-03-27 23:14:25.000000000 +0000
+++ grub-0.97-syllable/lib/device.c	2006-08-30 14:08:33.000000000 +0000
@@ -306,6 +306,16 @@
 #elif defined(__QNXNTO__)
   /* QNX RTP */
   sprintf (name, "/dev/fd%d", unit);
+#elif defined(__SYLLABLE__)
+  /* Syllable */
+# if 0
+  /* XXXKV: In theory this is simple enough, but GRUB uses an ioctl() on the device that
+     causes the driver to crash.  Don't re-enable this until the driver is fixed. */ 
+  sprintf (name, "/dev/disk/bios/fd%c/raw", 'a' + unit );*/
+# else
+# warning "BIOS floppy drives cannot be probed on Syllable."
+  *name = 0;
+# endif
 #else
 # warning "BIOS floppy drives cannot be guessed in your operating system."
   /* Set NAME to a bogus string.  */
@@ -347,6 +357,9 @@
   /* Actually, QNX RTP doesn't distinguish IDE from SCSI, so this could
      contain SCSI disks.  */
   sprintf (name, "/dev/hd%d", unit);
+#elif defined(__SYLLABLE__)
+  /* Syllable */
+  sprintf (name, "/dev/disk/ata/hd%c/raw", 'a' + unit );
 #else
 # warning "BIOS IDE drives cannot be guessed in your operating system."
   /* Set NAME to a bogus string.  */
@@ -388,6 +401,9 @@
   /* QNX RTP doesn't distinguish SCSI from IDE, so it is better to
      disable the detection of SCSI disks here.  */
   *name = 0;
+#elif defined(__SYLLABLE__)
+  /* Syllable */
+  sprintf (name, "/dev/disk/scsi/hd%c/raw", 'a' + unit );
 #else
 # warning "BIOS SCSI drives cannot be guessed in your operating system."
   /* Set NAME to a bogus string.  */
diff -ru grub-0.97/stage1/Makefile.am grub-0.97-syllable/stage1/Makefile.am
--- grub-0.97/stage1/Makefile.am	2004-07-16 11:44:56.000000000 +0000
+++ grub-0.97-syllable/stage1/Makefile.am	2006-08-30 14:08:33.000000000 +0000
@@ -2,6 +2,7 @@
 nodist_pkglib_DATA = stage1
 
 CLEANFILES = $(nodist_pkglib_DATA)
+AM_CFLAGS = -specs=../bodge-specs
 
 # We can't use builtins or standard includes.
 AM_CCASFLAGS = $(STAGE1_CFLAGS) -fno-builtin -nostdinc
diff -ru grub-0.97/stage2/Makefile.am grub-0.97-syllable/stage2/Makefile.am
--- grub-0.97/stage2/Makefile.am	2006-08-30 14:09:03.000000000 +0000
+++ grub-0.97-syllable/stage2/Makefile.am	2006-08-30 14:08:33.000000000 +0000
@@ -1,3 +1,5 @@
+AM_CFLAGS = -specs=../bodge-specs
+
 # For test target.
 TESTS = size_test
 noinst_SCRIPTS = $(TESTS)
