diff -ru lfs-bootscripts-6.3-original/lfs/init.d/functions lfs-bootscripts-6.3/lfs/init.d/functions
--- lfs-bootscripts-6.3-original/lfs/init.d/functions	2007-04-25 01:16:37.000000000 +0200
+++ lfs-bootscripts-6.3/lfs/init.d/functions	2007-09-03 17:57:22.000000000 +0200
@@ -16,7 +16,7 @@
 ## Environmental setup
 # Setup default values for environment
 umask 022
-export PATH="/bin:/usr/bin:/sbin:/usr/sbin"
+export PATH="/usr/indexes/sbin:/usr/indexes/bin:/usr/sbin:/usr/bin:/sbin:/bin:/system/indexes/sbin:/system/indexes/bin:/system/bin"
 
 # Signal sent to running processes to refresh their configuration
 RELOADSIG="HUP"
