diff -ru BinaryUtilities-2.22-original/bfd/config.bfd BinaryUtilities-2.22/bfd/config.bfd
--- BinaryUtilities-2.22-original/bfd/config.bfd	2011-07-29 00:35:13.000000000 +0200
+++ BinaryUtilities-2.22/bfd/config.bfd	2012-05-18 17:10:57.000000000 +0200
@@ -645,6 +645,9 @@
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs="i386pe_vec i386pei_vec"
     ;;
+  i[3-7]86-*-syllable*)
+    targ_defvec=bfd_elf32_i386_vec
+    ;;
   i[3-7]86-*-interix*)
     targ_defvec=i386pei_vec
     targ_selvecs="i386pe_vec"
diff -ru BinaryUtilities-2.22-original/gas/configure.tgt BinaryUtilities-2.22/gas/configure.tgt
--- BinaryUtilities-2.22-original/gas/configure.tgt	2011-06-20 15:23:21.000000000 +0200
+++ BinaryUtilities-2.22/gas/configure.tgt	2012-05-18 17:10:57.000000000 +0200
@@ -173,6 +173,7 @@
   i386-sequent-bsd*)			fmt=aout em=dynix ;;
   i386-*-beospe*)			fmt=coff em=pe ;;
   i386-*-beos*)				fmt=elf ;;
+  i386-*-syllable* )		fmt=elf bfd_gas=yes ;;
   i386-*-coff)				fmt=coff ;;
   i386-*-elf)				fmt=elf ;;
   i386-*-kaos*)				fmt=elf ;;
diff -ru BinaryUtilities-2.22-original/ld/Makefile.am BinaryUtilities-2.22/ld/Makefile.am
--- BinaryUtilities-2.22-original/ld/Makefile.am	2011-07-22 22:22:37.000000000 +0200
+++ BinaryUtilities-2.22/ld/Makefile.am	2012-05-18 17:10:57.000000000 +0200
@@ -257,6 +257,7 @@
 	eelf32xtensa.c \
 	eelf_i386.c \
 	eelf_i386_be.c \
+	eelf_i386_syllable.c \
 	eelf_i386_chaos.c \
 	eelf_i386_fbsd.c \
 	eelf_i386_ldso.c \
@@ -1177,6 +1178,9 @@
 eelf_i386_be.c: $(srcdir)/emulparams/elf_i386_be.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386_be "$(tdir_elf_i386_be)"
+eelf_i386_syllable.c: $(srcdir)/emulparams/elf_i386_syllable.sh \
+  $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+	${GENSCRIPTS} elf_i386_syllable "$(tdir_elf_i386_syllable)"
 eelf_i386_chaos.c: $(srcdir)/emulparams/elf_i386_chaos.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf_chaos.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386_chaos "$(tdir_elf_i386_chaos)"
diff -ru BinaryUtilities-2.22-original/ld/configure.tgt BinaryUtilities-2.22/ld/configure.tgt
--- BinaryUtilities-2.22-original/ld/configure.tgt	2011-11-21 10:29:37.000000000 +0100
+++ BinaryUtilities-2.22/ld/configure.tgt	2012-05-18 17:10:57.000000000 +0200
@@ -276,6 +276,7 @@
  			targ_extra_ofiles="deffilep.o pe-dll.o" ;;
 i[3-7]86-*-beospe*)	targ_emul=i386beos ;;
 i[3-7]86-*-beos*)	targ_emul=elf_i386_be ;;
+i[3-7]86-*-syllable*)	targ_emul=elf_i386_syllable ;;
 i[3-7]86-*-vxworks*)	targ_emul=elf_i386_vxworks ;;
 i[3-7]86-*-chaos)	targ_emul=elf_i386_chaos
 			;;
