diff -ruN binutils-2.10/bfd/config.bfd ../AtheOS/ports/src/binutils/bfd/config.bfd
--- binutils-2.10/bfd/config.bfd	2000-05-13 12:23:46.000000000 +0000
+++ ../AtheOS/ports/src/binutils/bfd/config.bfd	2000-10-25 17:03:42.000000000 +0000
@@ -329,6 +329,10 @@
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs="i386pe_vec i386pei_vec"
     ;;
+  i[3456]86-*-atheos*)
+    targ_defvec=bfd_elf32_i386_vec
+    targ_selvecs="i386pe_vec i386pei_vec"
+    ;;
   i[3456]86-*-interix*)
     targ_defvec=i386pei_vec
     targ_selvecs="i386pe_vec"
diff -ruN binutils-2.10/gas/configure ../AtheOS/ports/src/binutils/gas/configure
--- binutils-2.10/gas/configure	2000-05-31 11:28:42.000000000 +0000
+++ ../AtheOS/ports/src/binutils/gas/configure	2000-10-25 17:05:58.000000000 +0000
@@ -1712,6 +1712,7 @@
       i386-sequent-bsd*)    fmt=aout em=dynix bfd_gas=yes ;;
       i386-*-beospe*)       fmt=coff em=pe bfd_gas=yes ;;
       i386-*-beoself* | i386-*-beos*) fmt=elf bfd_gas=yes ;;
+      i386-*-atheos*)       fmt=elf bfd_gas=yes ;;
       i386-*-bsd*)          fmt=aout em=386bsd ;;
       i386-*-netbsd0.8)     fmt=aout em=386bsd ;;
       i386-*-netbsd*)       fmt=aout em=nbsd bfd_gas=yes;;
diff -ruN binutils-2.10/gas/configure.in ../AtheOS/ports/src/binutils/gas/configure.in
--- binutils-2.10/gas/configure.in	2000-05-31 11:28:43.000000000 +0000
+++ ../AtheOS/ports/src/binutils/gas/configure.in	2000-10-25 17:05:59.000000000 +0000
@@ -190,6 +190,7 @@
       i386-sequent-bsd*)    fmt=aout em=dynix bfd_gas=yes ;;
       i386-*-beospe*)       fmt=coff em=pe bfd_gas=yes ;;
       i386-*-beoself* | i386-*-beos*) fmt=elf bfd_gas=yes ;;
+      i386-*-atheos*)       fmt=elf bfd_gas=yes ;;
       i386-*-bsd*)          fmt=aout em=386bsd ;;
       i386-*-netbsd0.8)     fmt=aout em=386bsd ;;
       i386-*-netbsd*)       fmt=aout em=nbsd bfd_gas=yes;;
diff -ruN binutils-2.10/ld/Makefile.am ../AtheOS/ports/src/binutils/ld/Makefile.am
--- binutils-2.10/ld/Makefile.am	2000-04-05 05:36:44.000000000 +0000
+++ ../AtheOS/ports/src/binutils/ld/Makefile.am	2000-10-25 17:12:28.000000000 +0000
@@ -141,6 +141,7 @@
 	eelf32ppclinux.o \
 	eelf_i386.o \
 	eelf_i386_be.o \
+	eelf_i386_atheos.o \
 	egld960.o \
 	egld960coff.o \
 	eelf32fr30.o \
@@ -450,6 +451,9 @@
 eelf_i386_be.c: $(srcdir)/emulparams/elf_i386_be.sh \
   $(srcdir)/emultempl/elf32.em $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386_be "$(tdir_elf_i386_be)"
+eelf_i386_atheos.c: $(srcdir)/emulparams/elf_i386_atheos.sh \
+  $(srcdir)/emultempl/elf32.em $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+	${GENSCRIPTS} elf_i386_atheos "$(tdir_elf_i386_atheos)"
 egld960.c: $(srcdir)/emulparams/gld960.sh \
   $(srcdir)/emultempl/gld960.em $(srcdir)/scripttempl/i960.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} gld960 "$(tdir_gld960)"
diff -ruN binutils-2.10/ld/Makefile.in ../AtheOS/ports/src/binutils/ld/Makefile.in
--- binutils-2.10/ld/Makefile.in	2000-04-05 05:36:44.000000000 +0000
+++ ../AtheOS/ports/src/binutils/ld/Makefile.in	2000-10-25 17:12:29.000000000 +0000
@@ -244,6 +244,7 @@
 	eelf32ppclinux.o \
 	eelf_i386.o \
 	eelf_i386_be.o \
+	eelf_i386_atheos.o \
 	egld960.o \
 	egld960coff.o \
 	eelf32fr30.o \
@@ -1149,6 +1150,9 @@
 eelf_i386_be.c: $(srcdir)/emulparams/elf_i386_be.sh \
   $(srcdir)/emultempl/elf32.em $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386_be "$(tdir_elf_i386_be)"
+eelf_i386_atheos.c: $(srcdir)/emulparams/elf_i386_atheos.sh \
+  $(srcdir)/emultempl/elf32.em $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+	${GENSCRIPTS} elf_i386_atheos "$(tdir_elf_i386_atheos)"
 egld960.c: $(srcdir)/emulparams/gld960.sh \
   $(srcdir)/emultempl/gld960.em $(srcdir)/scripttempl/i960.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} gld960 "$(tdir_gld960)"
diff -ruN binutils-2.10/ld/configure.tgt ../AtheOS/ports/src/binutils/ld/configure.tgt
--- binutils-2.10/ld/configure.tgt	2000-05-28 11:32:35.000000000 +0000
+++ ../AtheOS/ports/src/binutils/ld/configure.tgt	2000-10-25 17:12:30.000000000 +0000
@@ -103,6 +103,7 @@
 i[3456]86-*-rtemscoff*) targ_emul=i386coff ;;
 i[3456]86-*-rtemself*)  targ_emul=elf_i386 ;;
 i[3456]86-*-rtems*)     targ_emul=elf_i386 ;;
+i[3456]86-*-atheos*)	targ_emul=elf_i386_atheos ;;
 i[3456]86-*-bsd)	targ_emul=i386bsd ;;
 i[3456]86-*-bsd386)	targ_emul=i386bsd ;;
 i[3456]86-*-bsdi*)	targ_emul=i386bsd ;;
diff -ruN binutils-2.10/ld/emulparams/elf_i386_atheos.sh ../AtheOS/ports/src/binutils/ld/emulparams/elf_i386_atheos.sh
--- binutils-2.10/ld/emulparams/elf_i386_atheos.sh	1970-01-01 00:00:01.000000000 +0000
+++ ../AtheOS/ports/src/binutils/ld/emulparams/elf_i386_atheos.sh	2000-10-25 17:14:27.000000000 +0000
@@ -0,0 +1,11 @@
+SCRIPT_NAME=elf
+OUTPUT_FORMAT="elf32-i386"
+TEXT_START_ADDR=0x80000000
+SHLIB_TEXT_START_ADDR=0x80000000
+MAXPAGESIZE=0x1000
+NONPAGED_TEXT_START_ADDR=0x80000000
+ARCH=i386
+MACHINE=
+NOP=0x9090
+TEMPLATE_NAME=elf32
+GENERATE_SHLIB_SCRIPT=yes
